sdk/core/src/http_client.rs

```diff
@@ -1,11 +1,12 @@
-use crate::errors::{AzureError, UnexpectedHTTPResult};
+use crate::errors::*;
 use async_trait::async_trait;
 use bytes::Bytes;
 use http::{Request, Response, StatusCode};
+#[cfg(feature = "enable_hyper")]
 use hyper_rustls::HttpsConnector;
 use serde::Serialize;
 
-#[async_trait]
+#[async_trait(?Send)]
```

Comment:
Can you explain why this is necessary/desired?

Reply:
I explained in https://github.com/Azure/azure-sdk-for-rust/pull/228#issuecomment-817344849. If I leave it as is, I get this error:
```
Rc<RefCell<wasm_bindgen_futures::Inner>>` cannot be sent between threads safely
```

Reply:
I'm not sure it if possible to mark the clients as `Send` for non-wasm builds. This sucks. I'm not sure how to proceed. I really want wasm support.